pkg/prometheus/promcfg.go

```diff
@@ -1691,6 +1691,8 @@ func (cg *ConfigGenerator) generateRemoteReadConfig(
 			cfg = append(cfg, yaml.MapItem{Key: "proxy_url", Value: spec.ProxyURL})
 		}
 
+		cfg = append(cfg, yaml.MapItem{Key: "filter_external_labels", Value: spec.FilterExternalLabels})
```

Comment:
There needs to be a check of the Prometheus version

```suggestion
		cfg = cg.WithMinimumVersion("2.34.0").AppendMapItem(cfg, yaml.MapItem{Key: "filter_external_labels", Value: spec.FilterExternalLabels})
```